code/game/objects/effects/spawners/detgun_spawner.dm

```diff
@@ -0,0 +1,12 @@
+/obj/effect/spawner/detgun
+	name = "detective's gun spawner"
+	icon = 'icons/effects/spawner_icons.dmi'
+	icon_state = "questionmark"
+
+/obj/effect/spawner/detgun/Initialize(mapload)
+	. = ..()
+	var/obj/item/gun/energy/detective/detgun = new /obj/item/gun/energy/detective(get_turf(src))
+	var/obj/item/pinpointer/crew/pointer = new /obj/item/pinpointer/crew(get_turf(src))
```

Comment:
won't doing `get_turf` mean this won't spawn in lockers?

Reply:
worked during testing, I suppose locker contents are spawned before scooping of the turf content